### PR TITLE
[SeaTunnel#989]Optimize maven config.

### LINKED ISSUE #989: [Bug] [seatunnel-dist] dev branch dist commit job failed
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-seatunnel/issues?q=is%3Aissue+label%3A%22bug%22) and found no similar issues.

### What happened

exec /Users/artefact/Downloads/flink-1.9.3/bin/flink run -c org.apache.seatunnel.SeatunnelFlink /Users/artefact/Documents/seatunnel-dist-2.0.5-SNAPSHOT-2.11.8/lib/seatunnel-core-flink.jar --config ./config/flink.streaming.conf.template  
above is running cmd，but seatunnel-core-flink.jar not contains seatunnel-core-base.jar

### SeaTunnel Version

2.X

### SeaTunnel Config

```conf
flink.streaming.conf.template
```

### Running Command

```shell
sh -x bin/start-seatunnel-flink.sh --config ./config/flink.streaming.conf.template
```

### Error Exception

```log
java.lang.NoClassDefFoundError: org/apache/seatunnel/config/command/CommandLineUtils
	at org.apache.seatunnel.SeatunnelFlink.main(SeatunnelFlink.java:28)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.apache.flink.client.program.PackagedProgram.callMainMethod(PackagedProgram.java:604)
	at org.apache.flink.client.program.PackagedProgram.invokeInteractiveModeForExecution(PackagedProgram.java:466)
	at org.apache.flink.client.program.ClusterClient.run(ClusterClient.java:274)
	at org.apache.flink.client.cli.CliFrontend.executeProgram(CliFrontend.java:746)
	at org.apache.flink.client.cli.CliFrontend.runProgram(CliFrontend.java:273)
	at org.apache.flink.client.cli.CliFrontend.run(CliFrontend.java:205)
	at org.apache.flink.client.cli.CliFrontend.parseParameters(CliFrontend.java:1008)
	at org.apache.flink.client.cli.CliFrontend.lambda$main$10(CliFrontend.java:1081)
	at org.apache.flink.runtime.security.NoOpSecurityContext.runSecured(NoOpSecurityContext.java:30)
	at org.apache.flink.client.cli.CliFrontend.main(CliFrontend.java:1081)
Caused by: java.lang.ClassNotFoundException: org.apache.seatunnel.config.command.CommandLineUtils
	at java.net.URLClassLoader.findClass(URLClassLoader.java:382)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at org.apache.flink.util.ChildFirstClassLoader.loadClass(ChildFirstClassLoader.java:69)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	... 15 more
```

### Flink or Spark Version

FLINK1.9

### Java or Scala Version

java8

### Screenshots

image

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by dongzl**
## Purpose of this pull request

this closes #989 
## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -757,12 +757,6 @@
                 </plugin>
                 <!-- maven deploy (End) -->
 
-                <plugin>
```
Comment (leo65535) [inline pom.xml:760]: 👍 remove the duplicate maven plugin config.

this is the cause of https://github.com/apache/incubator-seatunnel/issues/989, acked.

File: seatunnel-examples/seatunnel-flink-examples/pom.xml [hunk 2]
```diff
@@ -35,7 +35,7 @@
         <dependency>
             <groupId>org.apache.seatunnel</groupId>
             <artifactId>seatunnel-core-flink</artifactId>
-            <version>${parent.version}</version>
+            <version>${project.parent.version}</version>
```
Comment (leo65535) [inline seatunnel-examples/seatunnel-flink-examples/pom.xml:38]: please revert this change, duplicate to https://github.com/apache/incubator-seatunnel/pull/994

Comment (dongzl) [reply seatunnel-examples/seatunnel-flink-examples/pom.xml]: Thanks, revert.

**Review by leo65535 — APPROVED** [review]
thanks, 👍.

nit: please rename the tile of the patch, e.g `[feature / improve] Optimize mvn config`

**Review by CalvinKirs — APPROVED** [review]
LGTM
